Debug only

UnaKarlsen opened the pull request:
Added a debug-only mode that enables debug mode without collecting logs.

UnaKarlsen committed:
Added a debug-only mode, allowing users to enable debug mode without collecting logs.
Pack was also moved to be run in the brief and debug sections of the mode case statement rather than at the end of the script.

UnaKarlsen committed:
Created init() and moved the execution of is_root and get_sysinfo there as it is common to collect_brief and enable_debug.

UnaKarlsen committed:
Fixed Typo

jhaynes commented on ecs-logs-collector.sh:
```diff
@@ -167,12 +173,17 @@ collect_brief() {
   enable_ecs_agent_debug
 }
 
+collect_debug() {
```
This function is kinda misleading now since it doesn't actually collect debug logs. I think I'd vote for eliminating it altogether an just calling `collect_brief` and `enable_debug` in the `debug` case on line 538.

UnaKarlsen replied:
Makes sense. collect_debug() was the existing function name so I had left it that way, but removing it in favour of combining the collect and debug in the case statement does make it cleaner.

jhaynes commented on ecs-logs-collector.sh:
```diff
@@ -50,10 +50,13 @@ help()
+  echo "                 ECS container agent logs. This is the default mode."
+  echo "     debug       Collects 'brief' logs and also enables debug mode for the"
+  echo "                 Docker daemon and the Amazon ECS container agent."
+  echo "     debug-only  enables debug mode for the Docker daemon and the Amazon"
```
Nit: s/enables/Enables/

jhaynes reviewed (commented):
This looks great. Thanks. Can you add a little to your PR comment to indicate what testing you've done?

UnaKarlsen committed:
Updates as per PR

Fixed the case of Enables on line 57
Removed  collect_debug() in favour of calling collect_brief() and enable_debug() in the case statement at line 531

SmilyBorg commented:
I have updated the code as per comments by @jhaynes 

I have tested the updated script against amzn-ami-2017.03.f-amazon-ecs-optimized in all 3 modes with both debug enabled and disabled prior to execution. All results were as expected.
